File: bridge/typescript/text-message.ts
import {Message} from "./message";
import {IMessageSender} from "./message-sender.interface";

// UML diyagramındaki RefinedAbstraction kısmına denk gelmektedir.
// Abstraction (Message) sınıfından türer.
export class TextMessage extends Message {
  constructor(protected readonly messageSender: IMessageSender) {
    super(messageSender);
  }

  public send(): void {
   this.messageSender.sendMessage(this);
  }
}

File: bridge/go/text-message.go
package main

type TextMessage struct {
	messageSender MessageSender
}

func NewTextMessage(messageSender MessageSender) *TextMessage {
	return &TextMessage{
		messageSender: messageSender,
	}
}

func (m *TextMessage) Send() {
	m.messageSender.sendMessage(m)
}
